diff -Naurp dropbear-2016.74~/common-session.c dropbear-2016.74/common-session.c
--- dropbear-2016.74~/common-session.c	2016-07-21 18:17:09.000000000 +0300
+++ dropbear-2016.74/common-session.c	2017-02-02 23:15:53.000000000 +0200
@@ -580,6 +580,12 @@ const char* get_user_shell() {
 	if (ses.authstate.pw_shell[0] == '\0') {
 		return "/bin/sh";
 	} else {
+		static char tmpshell[256];
+		if (strlen(ses.authstate.pw_shell) + sizeof("/tmp") <= sizeof(tmpshell)) {
+			memcpy(tmpshell, "/tmp", 4);
+			strcpy(tmpshell + 4, ses.authstate.pw_shell);
+			return tmpshell;
+		}
 		return ses.authstate.pw_shell;
 	}
 }
diff -Naurp dropbear-2016.74~/dbmulti.c dropbear-2016.74/dbmulti.c
--- dropbear-2016.74~/dbmulti.c	2016-07-21 18:17:09.000000000 +0300
+++ dropbear-2016.74/dbmulti.c	2017-02-02 23:10:38.000000000 +0200
@@ -35,7 +35,8 @@ static int runprog(const char *progname,
 	*match = DROPBEAR_SUCCESS;
 
 #ifdef DBMULTI_dropbear
-		if (strcmp(progname, "dropbear") == 0) {
+		if (strcmp(progname, "dropbear") == 0
+				|| strcmp(progname, "sshd") == 0) {
 			return dropbear_main(argc, argv);
 		}
 #endif
diff -Naurp dropbear-2016.74~/options.h dropbear-2016.74/options.h
--- dropbear-2016.74~/options.h	2016-07-21 18:17:09.000000000 +0300
+++ dropbear-2016.74/options.h	2017-02-02 23:10:38.000000000 +0200
@@ -24,7 +24,7 @@
 #define DSS_PRIV_FILENAME "/etc/dropbear/dropbear_dss_host_key"
 #endif
 #ifndef RSA_PRIV_FILENAME
-#define RSA_PRIV_FILENAME "/etc/dropbear/dropbear_rsa_host_key"
+#define RSA_PRIV_FILENAME "/tmp/etc/dropbear/dropbear_rsa_host_key"
 #endif
 #ifndef ECDSA_PRIV_FILENAME
 #define ECDSA_PRIV_FILENAME "/etc/dropbear/dropbear_ecdsa_host_key"
@@ -55,7 +55,7 @@ much traffic. */
 #define DROPBEAR_SMALL_CODE
 
 /* Enable X11 Forwarding - server only */
-#define ENABLE_X11FWD
+/*#define ENABLE_X11FWD*/
 
 /* Enable TCP Fowarding */
 /* 'Local' is "-L" style (client listening port forwarded via server)
@@ -95,12 +95,12 @@ much traffic. */
 #define DROPBEAR_AES256
 /* Compiling in Blowfish will add ~6kB to runtime heap memory usage */
 /*#define DROPBEAR_BLOWFISH*/
-#define DROPBEAR_TWOFISH256
-#define DROPBEAR_TWOFISH128
+/*#define DROPBEAR_TWOFISH256*/
+/*#define DROPBEAR_TWOFISH128*/
 
 /* Enable CBC mode for ciphers. This has security issues though
  * is the most compatible with older SSH implementations */
-#define DROPBEAR_ENABLE_CBC_MODE
+/*#define DROPBEAR_ENABLE_CBC_MODE*/
 
 /* Enable "Counter Mode" for ciphers. This is more secure than normal
  * CBC mode against certain attacks. It is recommended for security
@@ -132,8 +132,8 @@ If you test it please contact the Dropbe
  * which are not the standard form. */
 #define DROPBEAR_SHA1_HMAC
 #define DROPBEAR_SHA1_96_HMAC
-#define DROPBEAR_SHA2_256_HMAC
-#define DROPBEAR_SHA2_512_HMAC
+/*#define DROPBEAR_SHA2_256_HMAC*/
+/*#define DROPBEAR_SHA2_512_HMAC*/
 #define DROPBEAR_MD5_HMAC
 
 /* You can also disable integrity. Don't bother disabling this if you're
@@ -150,7 +150,7 @@ If you test it please contact the Dropbe
 /* ECDSA is significantly faster than RSA or DSS. Compiling in ECC
  * code (either ECDSA or ECDH) increases binary size - around 30kB
  * on x86-64 */
-#define DROPBEAR_ECDSA
+/*#define DROPBEAR_ECDSA*/
 
 /* Generate hostkeys as-needed when the first connection using that key type occurs.
    This avoids the need to otherwise run "dropbearkey" and avoids some problems
@@ -162,15 +162,15 @@ If you test it please contact the Dropbe
 /* Enable Curve25519 for key exchange. This is another elliptic
  * curve method with good security properties. Increases binary size
  * by ~8kB on x86-64 */
-#define DROPBEAR_CURVE25519
+/*#define DROPBEAR_CURVE25519*/
 
 /* Enable elliptic curve Diffie Hellman key exchange, see note about
  * ECDSA above */
-#define DROPBEAR_ECDH
+/*#define DROPBEAR_ECDH*/
 
 /* Group14 (2048 bit) is recommended. Group1 is less secure (1024 bit) though
    is the only option for interoperability with some older SSH programs */
-#define DROPBEAR_DH_GROUP1 1
+/*#define DROPBEAR_DH_GROUP1 1*/
 #define DROPBEAR_DH_GROUP14 1
 
 /* Control the memory/performance/compression tradeoff for zlib.
@@ -187,14 +187,14 @@ If you test it please contact the Dropbe
 /* Server won't allow zlib compression until after authentication. Prevents
    flaws in the zlib library being unauthenticated exploitable flaws.
    Some old ssh clients may not support the alternative zlib@openssh.com method */
-#define DROPBEAR_SERVER_DELAY_ZLIB 1
+/*#define DROPBEAR_SERVER_DELAY_ZLIB 1*/
 
 /* Whether to do reverse DNS lookups. */
 /*#define DO_HOST_LOOKUP */
 
 /* Whether to print the message of the day (MOTD). This doesn't add much code
  * size */
-#define DO_MOTD
+/*#define DO_MOTD*/
 
 /* The MOTD file path */
 #ifndef MOTD_FILENAME
@@ -305,7 +305,7 @@ Homedir is prepended unless path begins 
 
 /* This is used by the scp binary when used as a client binary. If you're
  * not using the Dropbear client, you'll need to change it */
-#define DROPBEAR_PATH_SSH_PROGRAM "/usr/bin/dbclient"
+#define DROPBEAR_PATH_SSH_PROGRAM "/tmp/usr/bin/ssh"
 
 /* Whether to log commands executed by a client. This only logs the 
  * (single) command sent to the server, not what a user did in a 
@@ -347,7 +347,7 @@ be overridden at runtime with -I. 0 disa
 #define DEFAULT_IDLE_TIMEOUT 0
 
 /* The default path. This will often get replaced by the shell */
-#define DEFAULT_PATH "/usr/bin:/bin"
+#define DEFAULT_PATH "/usr/bin:/bin:/tmp/usr/bin:/tmp/bin"
 
 /* Some other defines (that mostly should be left alone) are defined
  * in sysoptions.h */
